watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1199,8 +1240,135 @@ def rule_visc_k_phase(b, p):
     def _radius_stokes_comp(self):
         add_object_reference(self, "radius_stokes_comp", self.params.radius_stokes_comp)
 
+    def _molar_volume_phase_comp(self):
+        add_object_reference(
+            self, "molar_volume_phase_comp", self.params.molar_volume_phase_comp
+        )
+
     def _diffus_phase_comp(self):
-        add_object_reference(self, "diffus_phase_comp", self.params.diffus_phase_comp)
+        # Retrieve two components sets from the molar_volume_data and diffusivity_data configurations
+        self.mv_dt_ind = Set(
```

Comment:
```suggestion
        self.molar_volume_data_index = Set(
```
It's longer and seems less wieldy, but it is also more readable more users/developers and we try to take that route whenever possible.